case_utils/case_validate/__init__.py

```diff
@@ -64,13 +65,175 @@ class NonExistentCDOConceptWarning(UserWarning):
     pass
 
 
+class ValidationResult:
+    conforms: bool
+    graph: Any
+    text: str
+    undefined_concepts: Set[rdflib.URIRef]
+
+
 def concept_is_cdo_concept(n_concept: rdflib.URIRef) -> bool:
+    """
+    Determine if a concept is part of the CDO ontology.
+
+    :param n_concept: The concept to check.
+    :return: whether the concept is part of the CDO ontology.
+    """
     concept_iri = str(n_concept)
     return concept_iri.startswith(
         "https://ontology.unifiedcyberontology.org/"
     ) or concept_iri.startswith("https://ontology.caseontology.org/")
 
 
+def get_ontology_graph(
+    case_version: Optional[str] = None, supplemental_graphs: Optional[List[str]] = None
+) -> rdflib.Graph:
+    """
+    Get the ontology graph for the given case_version and any supplemental graphs.
+
+    :param case_version: the version of the CASE ontology to use.  If None, the most recent version will be used.
```

Comment:
Bug: The implementation of this function doesn't quite match this specification.  If `case_version` is `None`, currently no CASE version would be loaded.

Reply:
Fixed

Reply:
Ah, another bug, and this one is due to a spelling/audial-verbalization issue.

These are supposed to have different behaviors:

```python
case_version = None
# vs.
case_version = "none"
```

If `None` is passed, this is assumed to have flowed from the CLI arguments not having `--case-version` specified.  In this case, the default value should be assumed, `CURRENT_CASE_VERSION`, as you have it.

If `"none"` is passed, this is assumed to have flowed from the CLI arguments having `--case-version=none` specified.  *This* value is intended to be an override, instructing the validator to NOT load any pre-built CASE/UCO/CDO ontologies.  All OWL+SHACL definitions are assumed to be brought in via the `supplemental_graphs` argument.  The use cases this supports are:

1. Use of `case_validate` against non-CDO graphs.
2. Overriding any CASE builds with a backwards-incompatible CASE build (e.g. the `*-2.0.0.ttl` files in [this directory](https://github.com/casework/casework.github.io/tree/master/ontology)).

I'll push a corrective patch.

Reply:
Fixed in `40be311`.